src/components/standup/FormInputComponent.tsx

```diff
@@ -1,36 +1,161 @@
+                                            required
+                                            value={inputValue}
+                                            onChange={(e) =>
+                                                handleInputChange(
+                                                    sectionIndex,
+                                                    inputIndex,
+                                                    e.target.value
+                                                )
+                                            }
+                                        />
+                                    </div>
+                                    <div className={styles.buttonsContainer}>
+                                        {inputIndex ===
+                                            section.inputs.length - 1 && (
+                                            <button
+                                                className={styles.addButton}
+                                                type="button"
+                                                onClick={() =>
+                                                    handleAddField(sectionIndex)
+                                                }
+                                            >
+                                                +
+                                            </button>
+                                        )}
+                                        {inputIndex ===
+                                            section.inputs.length - 1 &&
+                                            section.inputs.length > 1 && (
+                                                <button
+                                                    type="button"
+                                                    className={
+                                                        styles.removeButton
+                                                    }
+                                                    onClick={() =>
+                                                        handleRemoveField(
+                                                            sectionIndex
+                                                        )
+                                                    }
+                                                >
+                                                    x
+                                                </button>
+                                            )}
+                                    </div>
+                                </div>
+                            ))}
+                        </div>
+                    </div>
+                </div>
+            ))}
+            <button
+                type="submit"
+                disabled={!isFormValid()}
+                className={`${styles.submitButton}`}
+            >
+                Submit
+            </button>
+        </form>
```

Comment:
Can you break down this

Reply:
any solution you would like to suggest me?

Reply:
import React, { FC, useState } from 'react';
import { useRouter } from 'next/router';
import { useSaveProgressMutation } from '@/app/services/progressesApi';
import { toast, ToastTypes } from '@/helperFunctions/toast';
import StandupSection from './StandupSection';
import styles from '@/components/standup/standupContainer.module.scss';

const initialSections = [
  { title: 'Yesterday', inputs: [''] },
  { title: 'Today', inputs: [''] },
  { title: 'Blocker', inputs: [''] },
];

const FormInputComponent: FC = () => {
  const router = useRouter();
  const [sections, setSections] = useState(initialSections);
  const [addStandup] = useSaveProgressMutation();
  const { SUCCESS, ERROR } = ToastTypes;

  const handleAddField = (sectionIndex: number) => {
    const newSections = [...sections];
    newSections[sectionIndex].inputs.push('');
    setSections(newSections);
  };

  const handleRemoveField = (sectionIndex: number) => {
    const newSections = [...sections];
    newSections[sectionIndex].inputs.pop();
    setSections(newSections);
  };

  const handleInputChange = (
    sectionIndex: number,
    inputIndex: number,
    value: string
  ) => {
    const newSections = [...sections];
    newSections[sectionIndex].inputs[inputIndex] = value;
    setSections(newSections);
  };

  const isFormValid = () => {
    return sections.every((section) =>
      section.inputs.every((input) => input.trim() !== '')
    );
  };

  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    const newData = {
      type: 'user',
      completed: sections[0].inputs.join('. '),
      planned: sections[1].inputs.join('. '),
      blockers: sections[2].inputs.join('. '),
    };
    try {
      const data = await addStandup(newData).unwrap();
      toast(SUCCESS, data.message);
    } catch (error) {
      toast(ERROR, error.data.message);
    }
    setSections(initialSections);
    router.replace(router.asPath);
  };

  return (
    <form className={styles.standupForm} onSubmit={handleSubmit} aria-label="form">
      {sections.map((section, sectionIndex) => (
        <StandupSection
          key={sectionIndex}
          section={section}
          sectionIndex={sectionIndex}
          onInputChange={handleInputChange}
          onAddField={handleAddField}
          onRemoveField={handleRemoveField}
        />
      ))}
      <button type="submit" disabled={!isFormValid()} className={styles.submitButton}>
        Submit
      </button>
    </form>
  );
};

export default FormInputComponent;

import React from 'react';
import StandupInput from './StandupInput';

const StandupSection = ({
  section,
  sectionIndex,
  onInputChange,
  onAddField,
  onRemoveField,
}) => {
  return (
    <div className={styles.formFields}>
      <label className={styles.updateHeading} htmlFor={section.title}>
        {section.title}
      </label>
      <div className={styles.inputContainers}>
        <div className={styles.inputBoxContainer}>
          {section.inputs.map((inputValue, inputIndex) => (
            <StandupInput
              key={inputIndex}
              sectionIndex={sectionIndex}
              inputIndex={inputIndex}
              inputValue={inputValue}
              onInputChange={onInputChange}
              onAddField={onAddField}
              onRemoveField={onRemoveField}
              sectionTitle={section.title}
            />
          ))}
        </div>
      </div>
    </div>
  );
};

export default StandupSection;


import React from 'react';

const StandupInput = ({
  sectionIndex,
  inputIndex,
  inputValue,
  onInputChange,
  onAddField,
  onRemoveField,
  sectionTitle,
}) => {
  return (
    <div className={styles.inputAndButtons}>
      <div className={styles.inputBox}>
        <input
          className={styles.inputField}
          data-testid={`${sectionTitle}${inputIndex}`}
          name={sectionTitle}
          type="text"
          required
          value={inputValue}
          onChange={(e) => onInputChange(sectionIndex, inputIndex, e.target.value)}
        />
      </div>
      <div className={styles.buttonsContainer}>
        {inputIndex === section.inputs.length - 1 && (
          <button className={styles.addButton} type="button" onClick={() => onAddField(sectionIndex)}>
            +
          </button>
        )}
        {inputIndex === section.inputs.length - 1 && section.inputs.length > 1 && (
          <button className={styles.removeButton} type="button" onClick={() => onRemoveField(sectionIndex)}>
            x
          </button>
        )}
      </div>
    </div>
  );
};

export default StandupInput;